terraform/modules/ecr/main.tf

```diff
@@ -0,0 +1,40 @@
+terraform {
+  required_version = "~>1.5.1"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~>5.0"
+    }
+  }
+}
+
+#tfsec:ignore:aws-ecr-repository-customer-key:encryption using KMS CMK not required
+resource "aws_ecr_repository" "main" {
+  #checkov:skip=CKV_AWS_136:encryption using KMS not required
+  name                 = "core-ecr"
+  image_tag_mutability = "IMMUTABLE"
+
+  image_scanning_configuration {
+    scan_on_push = true
+  }
+}
+
+resource "aws_ecr_lifecycle_policy" "main" {
+  repository = aws_ecr_repository.main.name
+
+  policy = jsonencode({
+    rules = [{
+      rulePriority = 1
+      description  = "keep last 10 images"
+      action = {
+        type = "expire"
+      }
+      selection = {
+        tagStatus   = "any"
+        countType   = "imageCountMoreThan"
+        countNumber = 10
+      }
+    }]
+  })

```

Comment:
This needs thinking through as well - I expect to need to keep more than the last 10 images, since we're likely to build more than that between releases to production (particularly with review apps). 

I expect this is something to have a final answer for when thinking about tagging strategy and deployments (the ideal would probably be something clever that treats different envs differently), but for now I think it would be better to have nothing (because the consequences of not then changing it are better that way).

Reply:
Ah yes that makes sense. I've removed the policy all together for now and can add one in later once we have a clearer idea. Happy to make a ticket so we don't forget to make one later?

Reply:
There already is one in the deployment pipelines epic.